phi and K*0 are added in event record

sdudi123 opened the pull request:
The resonance particle (phi and K*0) are added in event record along with their daughter particle for starlight+dpmjet event production. Full chain of mother and daughter of resonance will be store in event record output.

Commit:
phi and K*0 are added in event record

sdudi123 commented:
Hi, @SpencerKlein  Could you please have a review of this?

SpencerKlein commented:
Hi,

This code is needed for a specific analysis, by Sandeep Dudi.  It is a modification of DPMJET to be able to output K^*0 and phi mesons, so as to associate MC final state particles with these daughters.  I completely support Sandeep doing this for his analysis.  However it is a rather specific modification, useful only for his analysis, choosing to output only the two Monte Carlo particles of interest for his analysis.  In a more general world, why not Lambdas or rho's or D^0 or ???  

There are downsides for these changes. They will increase the size of the output records.  Second, they may cause some people's analysis code to break if suddently phi and K^*0 start showing up in the output record.  

My preference would be to have this code controlled by a switch, so that it is only turned on when requested.  Ideally, there would be a set of switches allowing people to select which output particles that they would like, but a simpler approach would also be OK.
An alternate approach would be to keep this as a private version of STARlight/DPMJet.

Commit:
Logic switch are added for resonances

sdudi123 commented:
Hi, I have added the switch for resonances. I added phi, K*0, rho, and Lambda*. I think D0 are not produced in DPMJET. I can add more resonances if required. Thanks.

SpencerKlein commented:
Hi,

How do the variables KEEP_PHI, KEEP_KSTAR, KEEP_RHO0, KEEP_LAMBDASTAR get set?  The proposed new code says " Switches for each resonance (set in main program or defaults)"  I want to make sure that this code is not turned on automatically or accidentally for other users who do not want this functionality.

Spencer

sdudi123 commented:
Hi, I tried to add Switches in my.input file and run it locally, but it did't work. It required the corresponding change in src/dpmjet/DT_INIT.f . is it possible to keep separate dpmjetint.f with resonance ?

Commit:
phi and Kstar switch are added

sdudi123 commented:
Hi, @SpencerKlein ,  I have added the phi and K*0 switch through slight.in. Please have a look. Thanks.

Commit:
Phi and K*0 along with their daughters are store in starlight+dpmjet output

Commit:
Phi and K*0 along with their daughters are store in starlight+dpmjet output

Commit:
switch for phi and K*0 are added and their decay daughter are removed

Commit:
Phi and K*0 switch are added and documentation is done
